File: src/main/java/com/linebot/batch/NoticeTrainDelayBatch.java
package com.linebot.batch;

import com.linebot.service.notice.GetTrainDelayResourceService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Controller;

@Slf4j
@AllArgsConstructor
@Controller
public class NoticeTrainDelayBatch {
    private GetTrainDelayResourceService getTrainDelayResourceService;

    @Scheduled(cron = "0 */20 7-9 * * *", zone = "Asia/Tokyo")
    public void execute() {
        getTrainDelayResourceService.execute();
    }
}

File: src/main/kotlin/com/linebot/batch/NoticeTrainDelayBatch.kt
package com.linebot.batch

import com.linebot.service.notice.GetTrainDelayResourceService
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Controller

@Controller
class NoticeTrainDelayBatch(
        private val getTrainDelayResourceService: GetTrainDelayResourceService
) {

    @Scheduled(cron = "0 */20 7-9 * * *", zone = "Asia/Tokyo")
    fun execute() {
        getTrainDelayResourceService.execute()
    }
}
